src/inc/corprof.idl

```diff
@@ -2408,6 +2412,31 @@ interface ICorProfilerCallback8 : ICorProfilerCallback7
         [in] BOOL       fIsSafeToBlock);
 }
 
+[
+    object,
+    uuid(27583EC3-C8F5-482F-8052-194B8CE4705A),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback9 : ICorProfilerCallback8
+{
+    // This event is triggered whenever a dynamic method is garbage collected
+    // and subsequently unloaded.
+
+    HRESULT DynamicMethodUnloaded([in] FunctionID functionId);
+
+    HRESULT EventPipeEventDelivered(
+        [in] REFGUID eventGuid,
+        [in] DWORD eventId,
+        [in] DWORD eventVersion,
+        [in] DWORD eventThreadId,
+        [in] LARGE_INTEGER eventTimestamp,
```

Comment:
I was thinking about that and figured a profiler author would be able to call the necessary stuff on their end. But I'm happy to make it a FILETIME.